### ISSUE TITLE
Possible to add tok to idle_deauth and ndsctl_deauth?

### ISSUE THREAD

**Issue by monelgordillo** [openNDS/openNDS#13]
Is it possible to add the token to idle_deauth, ndsctl_deauth, etc?

I'm currently seeing these:

method
client mac
bytes incoming
bytes outgoing
session starts
session end

**Comment by bluewavenet** [comment]
@monelgordillo 
The token is just a random number generated only for the client session and is of no value in usage statistics.
Why do you need it?

**Comment by monelgordillo** [comment]
I'm trying to tie auth_client and idle_deauth together as one session.  There's the mac address in both auth_client & idle_deauth, but a mac address could have multiple different sessions.  I was thinking the token could be used as the session id.

**Comment by bluewavenet** [comment]
A client can not have concurrent sessions though, and client mac, session start, session end uniquely identify the session. So is there a need need to tie auth_client and idle_deauth together?

Now maybe in the case of a remote FAS in combination with BinAuth there might be a case for this......

I am testing secure level 3 with binauth at the moment and adding required extensions - so adding token as well might turn out to be not be too much extra effort.....

Convince me that it is needed....

**Comment by bluewavenet** [comment]
@monelgordillo 
I added it anyway :-D
Next release soon.

**Comment by monelgordillo** [comment]
@bluewavenet 

👍 I suppose I'm pretty convincing. :) Thank you!  I'm looking forward to the next release.  

Here was my original thought process:

I'm trying to tie a session, along with bytes incoming/outgoing, to a particular username.  I'm thinking like a shared tablet used by multiple people.  Currently, I can see username (along with other custom variables in redir) within auth_client in BinAuth log.  

Example BinAuth Log:

root@raspberrypi:~/fas# cat /run/ndslog/binauthlog.log 
Fri 22 May 19:46:24 BST 2020, New log file created
datetime=Tue 26 May 14:34:02 BST 2020, method=auth_client, clientmac=18:31:bf:8e:0d:11, clientip=10.3.141.148, **username=mon1**, password=na, redir=http://10.3.141.1:2050/status.html?acctsessionid=18:31:bf:8e:0d:11_1590500036, useragent=Mozilla%2f5.0%20%28Linux%3b%20Android%208.1.0%3b%20ASUS_X018D%20Build%2fO11019%3b%20wv%29%20AppleWebKit%2f537.36%20%28KHTML%2c%20like%20Gecko%29%20Version%2f4.0%20Chrome%2f83.0.4103.60%20Mobile%20Safari%2f537.36
datetime=Tue 26 May 14:34:02 BST 2020, method=client_auth, clientmac=18:31:bf:8e:0d:11, bytes_incoming=0, bytes_outgoing=0, session_start=0, session_end=8101259050733430377
datetime=Tue 26 May 14:45:50 BST 2020, method=idle_deauth, clientmac=18:31:bf:8e:0d:11, bytes_incoming=335253, bytes_outgoing=3884827, session_start=1590500042, session_end=8101259050733430377
datetime=Tue 26 May 15:28:08 BST 2020, method=auth_client, clientmac=18:31:bf:8e:0d:11, clientip=10.3.141.148, **username=mon2**, password=na, redir=http://10.3.141.1:2050/status.html?acctsessionid=18:31:bf:8e:0d:11_1590503284, useragent=Mozilla%2f5.0%20%28Linux%3b%20Android%208.1.0%3b%20ASUS_X018D%20Build%2fO11019%3b%20wv%29%20AppleWebKit%2f537.36%20%28KHTML%2c%20like%20Gecko%29%20Version%2f4.0%20Chrome%2f83.0.4103.60%20Mobile%20Safari%2f537.36
datetime=Tue 26 May 15:28:08 BST 2020, method=client_auth, clientmac=18:31:bf:8e:0d:11, bytes_incoming=0, bytes_outgoing=0, session_start=0, session_end=8101259050733430377
datetime=Tue 26 May 15:39:50 BST 2020, method=idle_deauth, clientmac=18:31:bf:8e:0d:11, bytes_incoming=358125, bytes_outgoing=3920580, session_start=1590503288, session_end=8101259050733430377

**Comment by bluewavenet** [comment]
@monelgordillo 
Here is a log output example from the new version (with url decoding of useragent done in the binauth script):
```
Tue May 26 07:29:23 UTC 2020, method=auth_client, clientmac=74:de:2b:b1:70:33, clientip=10.170.1.138, username=na, password=na, redir=http://onboard-wifi.net/, useragent=Mozilla/5.0 (X11; Linux x86_64; rv:60.0) Gecko/20100101 Firefox/60.0, token=4a89a8c6, custom=Custom data sent to BinAuth
Tue May 26 07:29:23 UTC 2020, method=client_auth, clientmac=74:de:2b:b1:70:33, bytes_incoming=0, bytes_outgoing=0, session_start=1590478163, session_end=1590564563, token=4a89a8c6

```

Note the following:
 * Session times are now set correctly
 * The variable token has been added
 * A new custom variable "custom=" has been added to auth_client. This can be up to 256 characters long and is sent from FAS in the query string.

**Comment by bluewavenet** [comment]
@monelgordillo 
This is implemented in v5.1.0beta, with a new release coming soon!

**Comment by bluewavenet** [comment]
@monelgordillo 
Version 5.1.0 is now released.
https://github.com/openNDS/openNDS/releases/tag/v5.1.0
